balancer/weightedroundrobin/balancer.go

```diff
@@ -309,24 +312,19 @@ func (b *wrrBalancer) NewSubConn(addrs []resolver.Address, opts balancer.NewSubC
 		oldListener(state)
 	}
 
-	sc, err := b.ClientConn.NewSubConn([]resolver.Address{addr}, opts)
-	if err != nil {
-		return sc, err
-	}
-
 	b.mu.Lock()
-	ewv, ok := b.addressWeights.Get(addr)
+	defer b.mu.Unlock()
+	ewi, ok := b.addressWeights.Get(addr)
 	if !ok {
-		b.mu.Unlock()
 		// SubConn state updates can come in for a no longer relevant endpoint
 		// weight (from the old system after a new config update is applied).
-		// Will eventually get cleared from scMap once receives Shutdown signal.
-		return sc, err
+		return nil, fmt.Errorf("balancer is being closed; no new SubConns allowed")
 	}
-	ew := ewv.(*endpointWeight)
-	b.scToWeight[sc] = ew
-	b.mu.Unlock()
-
+	sc, err := b.ClientConn.NewSubConn([]resolver.Address{addr}, opts)
+	if err != nil {
+		return nil, err
+	}
+	b.scToWeight[sc] = ewi.(*endpointWeight)
 	return sc, err
```

Comment:
`nil` explicitly, please, since you asserted `err == nil` already.

Reply:
Ah, done.